docs/developer.md

```diff
@@ -16,6 +16,7 @@ make presubmit
 kubectl apply -f config/crds/bases/k8s-gateway-v0.6.1.yaml
 kubectl apply -f config/crds/bases/multicluster.x-k8s.io_serviceexports.yaml
 kubectl apply -f config/crds/bases/multicluster.x-k8s.io_serviceimports.yaml
+kubectl apply -f config/crds/bases/externaldns.k8s.io_dnsendpoints.yaml
```

Comment:
Please also update "End-toEnd" testing section,  that e2e should run `OK` with both
* with DNS CRD
* `without`DNS CRD

Reply:
Actually I have bundled DNSEndpoint CRD with both helm chart and `deploy.yaml`, so it is just like other CRDs that we assume its existence. Though I have added missing CRD handling to assist local development, I don't think we have to force the setting for e2etest.

Reply:
Customer may choose to have a different way (other than the deploy.yaml or helm in github) to run lattice controller.  If we do not enforce testing it e2e and  miss a regression, it  can impact these customers

Reply:
Updated the doc